src/utils/clmfates_interfaceMod.F90

```diff
@@ -675,11 +686,30 @@ subroutine dynamics_driv(this, nc, bounds_clump,      &
                         model_day, floor(day_of_year), &
                         days_per_year, 1.0_r8/dble(days_per_year))
 
+      if (fates_spitfire_mode > 1) then
+         allocate(lnfm24(bounds_clump%begg:bounds_clump%endg), stat=ier)
+         if (ier /= 0) then
+            call endrun(msg="allocation error for lnfm24"//&
+                 errmsg(sourcefile, __LINE__))
+         endif
+
+         lnfm24 = this%fates_fire_data_method%GetLight24()
+      end if
 
       do s=1,this%fates(nc)%nsites
 
          c = this%f2hmap(nc)%fcolumn(s)
 
+         if (fates_spitfire_mode > 1) then
+            g = col%gridcell(c)
+            do ifp = 1, this%fates(nc)%sites(s)%youngest_patch%patchno
+               p = ifp + col%patchi(c)
+
+               this%fates(nc)%bc_in(s)%lightning24(ifp) = lnfm24(g) * 24._r8  ! #/km2/hr to #/km2/day
+               this%fates(nc)%bc_in(s)%pop_density(ifp) = this%fates_fire_data_method%forc_hdm(g)
```

Comment:
@ekluzek regarding my code mods to pass the population density to FATES:

I am comfortable with everything except line 709...
```
this%fates(nc)%bc_in(s)%pop_density(ifp) = this%fates_fire_data_method%forc_hdm(g)
```

I'm pointing it out in case you can tell that it won't work right off the bat. I plan to begin testing tomorrow, so I'll know soon enough :-)

Reply:
It looks correct to me, from just glancing at it...

Reply:
Thanks @ekluzek .